### PR TITLE
Add Announcement type and ForEachAnnouncement

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**

### CODE REVIEW

File: chain/announcements.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			var ha V1Announcement
+			dec := types.NewBufDecoder(arb)
+			ha.DecodeFrom(dec)
+			if err := dec.Err(); err != nil {
+				continue
+			} else if ha.Specifier != types.NewSpecifier(AnnouncementSpecifier) {
+				continue
+			}
+			// verify signature
+			if !ha.VerifySignature() {
```
Comment (peterjan) [inline chain/announcements.go:77]: `nit:` I'd turn this in an `else if` and join the validation?

File: chain/announcements.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			copy(pk[:], ha.PublicKey.Key)
+			fn(Announcement{
+				NetAddress: ha.NetAddress,
+				PublicKey:  pk,
+			})
+		}
+	}
+	for _, txn := range b.V2Transactions() {
+		for _, att := range txn.Attestations {
+			if att.Key != AnnouncementSpecifier {
```
Comment (peterjan) [inline chain/announcements.go:90]: should we add something similar to `VerifySignature` here or are attestations already verified somewhere?

**Review by peterjan — COMMENTED** [review]
LGTM 👍

File: chain/announcements.go (hunk 2 above)
Comment (ChrisSchinnerl) [reply chain/announcements.go:101]: `V2` attestations are already verified here

**Pull request merged**